tests/shells/mockspawn.py

```diff
@@ -0,0 +1,91 @@
+import logging
+from pexpect.spawnbase import (
+    SpawnBase,
+    TIMEOUT)
+from crl.interactivesessions.shells.remotemodules.chunkcomm import SharedBytesIO
+from .interpreter import (
+    Interpreter,
+    InterpreterTimeout)
+
+__copyright__ = 'Copyright (C) 2019, Nokia'
+
+
+class Default(object):
+    pass
+
+
+DEFAULT = Default()
+
+LOGGER = logging.getLogger(__name__)
+
+
+class MockSpawn(SpawnBase):
+
+    _handlers = None
+    _state = None
+
+    # pylint: disable=too-many-arguments
+    def __init__(self,
+                 command,
+                 args=None,
+                 timeout=DEFAULT,
+                 maxread=DEFAULT,
+                 searchwindowsize=DEFAULT,
+                 logfile=DEFAULT,
+                 env=None,
+                 ignore_sighup=False,
+                 encoding=DEFAULT,
+                 codec_errors=DEFAULT):
+        super(MockSpawn, self).__init__(**self._get_kwargs(
+            timeout=timeout,
+            maxread=maxread,
+            searchwindowsize=searchwindowsize,
+            logfile=logfile,
+            encoding=encoding,
+            codec_errors=codec_errors))
+        self._state.env = {} if env is None else env
+        self._state.ignore_sighup = ignore_sighup
+        self._interpreter = Interpreter(self._handlers, self._state)
```

Comment:
Copy of the static `_state` and `_handlers` should be used in the `__init__` to make testing two or more shells possible.

Reply:
This is a bit theoretical as there is usually no need to test more than one shell at the time.